falkordb/falkordb.py

```
@@ -225,6 +227,40 @@
+            replica connections.
+
+        Raises:
+            ConnectionError: If unable to connect or retrieve information from 
+            the FalkorDB setup.
+            ValueError: If the `mode` is neither Sentinel nor Cluster.
+        """
+        # decide if it's Sentinel or cluster
+        try:
+            mode = self.connection.execute_command("info")['redis_mode']
+        except Exception as e:
+            raise Exception(f"Failed to get Redis mode: {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)